core/arch/arm/plat-hikey/main.c

```diff
@@ -93,3 +108,166 @@ void console_flush(void)
 {
 	pl011_flush(console_base());
 }
+
+static vaddr_t peripherals_base(paddr_t pa)
+{
+	void *va;
+
+	if (cpu_mmu_enabled()) {
+		va = phys_to_virt(pa, MEM_AREA_IO_NSEC);
```

Comment:
you are right !
2 more small comments on this routine. 
- As this routine deals with nonsecure iomem only, you should maybe explicitly state it in the routine label (`nsec_periph_base()`).
- There is no real need for a local variable `va`.